diff --git a/de.fu_berlin.inf.dpp/ext-src/org/jivesoftware/smack/util/PacketParserUtils.java b/de.fu_berlin.inf.dpp/ext-src/org/jivesoftware/smack/util/PacketParserUtils.java
index 717d13c..4b17d26 100644
--- a/de.fu_berlin.inf.dpp/ext-src/org/jivesoftware/smack/util/PacketParserUtils.java
+++ b/de.fu_berlin.inf.dpp/ext-src/org/jivesoftware/smack/util/PacketParserUtils.java
@@ -706,7 +706,8 @@
             int eventType = parser.next();
 
             if (eventType == XmlPullParser.START_TAG) {
-                streamError = new StreamError(parser.getName());
+                if (streamError == null)
+                    streamError = new StreamError(parser.getName());
             } else if (eventType == XmlPullParser.END_TAG) {
                 if (parser.getName().equals("error")) {
                     done = true;