diff --git a/saros/ext-src/org/jivesoftware/smack/sasl/SASLGSSAPIMechanism.java b/saros/ext-src/org/jivesoftware/smack/sasl/SASLGSSAPIMechanism.java
index fb2ee89..c9a32bd 100644
--- a/saros/ext-src/org/jivesoftware/smack/sasl/SASLGSSAPIMechanism.java
+++ b/saros/ext-src/org/jivesoftware/smack/sasl/SASLGSSAPIMechanism.java
@@ -70,8 +70,7 @@
         String[] mechanisms = { getName() };
         Map<String, String> props = new HashMap<String, String>();
         props.put(Sasl.SERVER_AUTH, "TRUE");
-        sc = Sasl.createSaslClient(mechanisms, username, "xmpp", host, props,
-            cbh);
+        sc = Sasl.createSaslClient(mechanisms, null, "xmpp", host, props, cbh);
         authenticate();
     }
 
@@ -96,8 +95,7 @@
         String[] mechanisms = { getName() };
         Map<String, String> props = new HashMap<String, String>();
         props.put(Sasl.SERVER_AUTH, "TRUE");
-        sc = Sasl.createSaslClient(mechanisms, username, "xmpp", host, props,
-            this);
+        sc = Sasl.createSaslClient(mechanisms, null, "xmpp", host, props, this);
         authenticate();
     }
 
diff --git a/saros/ext-src/org/jivesoftware/smack/sasl/SASLMechanism.java b/saros/ext-src/org/jivesoftware/smack/sasl/SASLMechanism.java
index 4ea801f..2681b09 100644
--- a/saros/ext-src/org/jivesoftware/smack/sasl/SASLMechanism.java
+++ b/saros/ext-src/org/jivesoftware/smack/sasl/SASLMechanism.java
@@ -154,7 +154,7 @@
 
         String[] mechanisms = { getName() };
         Map<String, String> props = new HashMap<String, String>();
-        sc = Sasl.createSaslClient(mechanisms, username, "xmpp", serviceName,
+        sc = Sasl.createSaslClient(mechanisms, null, "xmpp", serviceName,
             props, this);
         authenticate();
     }
@@ -205,8 +205,7 @@
         throws IOException, XMPPException {
         String[] mechanisms = { getName() };
         Map<String, String> props = new HashMap<String, String>();
-        sc = Sasl.createSaslClient(mechanisms, username, "xmpp", host, props,
-            cbh);
+        sc = Sasl.createSaslClient(mechanisms, null, "xmpp", host, props, cbh);
         authenticate();
     }
 
