diff --git a/de.fu_berlin.inf.dpp/ext-src/org/jivesoftware/smack/packet/StreamError.java b/de.fu_berlin.inf.dpp/ext-src/org/jivesoftware/smack/packet/StreamError.java
index fb55741..49d7064 100644
--- a/de.fu_berlin.inf.dpp/ext-src/org/jivesoftware/smack/packet/StreamError.java
+++ b/de.fu_berlin.inf.dpp/ext-src/org/jivesoftware/smack/packet/StreamError.java
@@ -151,11 +151,19 @@
  */
 public class StreamError {
 
+    public static final String NAMESPACE = "urn:ietf:params:xml:ns:xmpp-streams";
+
     private String code;
+    private String text;
 
     public StreamError(String code) {
         super();
         this.code = code;
+    }
+
+    public StreamError(String code, String text) {
+        this(code);
+        this.text = text;
     }
 
     /**
@@ -167,10 +175,21 @@
         return code;
     }
 
+    /**
+     * Returns the error text, which may be null.
+     * 
+     * @return the error text.
+     */
+    public String getText() {
+        return text;
+    }
+
     @Override
     public String toString() {
         StringBuilder txt = new StringBuilder();
         txt.append("stream:error (").append(code).append(")");
+        if (text != null)
+            txt.append(" text: ").append(text);
         return txt.toString();
     }
 }