diff --git a/saros/ext-src/org/picocontainer/parameters/BasicComponentParameter.java b/saros/ext-src/org/picocontainer/parameters/BasicComponentParameter.java
index 055b443..bf13dcb 100644
--- a/saros/ext-src/org/picocontainer/parameters/BasicComponentParameter.java
+++ b/saros/ext-src/org/picocontainer/parameters/BasicComponentParameter.java
@@ -275,7 +275,7 @@
         if (container.getParent() != null) {
             if (binding != null) {
                 return container.getParent().getComponentAdapter(expectedType,
-                    binding.getClass());
+                    binding.annotationType());
             } else {
                 return container.getParent().getComponentAdapter(expectedType,
                     expectedNameBinding);
